src/oas3/accessors.ts

```diff
@@ -12,15 +12,34 @@ export function getSecurities(
   document: DeepPartial<OpenAPIObject>,
   operationSecurity?: OperationSecurities,
 ): SecurityWithKey[][] {
-  const opSchemesPairs = operationSecurity ? mapToKeys(operationSecurity) : mapToKeys(document.security);
+  const opSchemesPairs = operationSecurity
+    ? mapToKeys(operationSecurity)
+    : document.security
+    ? mapToKeys(document.security)
+    : [];
+  const flattenPairs: Dictionary<string[]> = operationSecurity
+    ? Object.assign({}, ...operationSecurity)
+    : document.security
+    ? Object.assign({}, ...document.security)
+    : {};
   const definitions = document.components?.securitySchemes;
 
   if (!isObject(definitions)) return [];
 
   return opSchemesPairs.map(opSchemePair =>
     opSchemePair
       .map(opScheme => {
-        return { ...definitions[opScheme], key: opScheme };
+        const definition = definitions[opScheme];
+
+        if (isSecurityScheme(definition) && definition.type === 'oauth2') {
+          // Put back only the flows that are part of the current definition
+          definition.flows = mapValues(definition.flows, (flow: OAuthFlowObject) => ({
```

Comment:
Hmm, aren't you mutating the original document here?

Reply:
Ah! Good point, I'm afraid I am. I'll fix that!

Reply:
Done in 9a4c8b4